lsp-pwsh.el

```diff
@@ -1,134 +1,139 @@
+  "Return the command to start server."
+  `(,lsp-pwsh-exe "-NoProfile" "-NonInteractive" "-NoLogo"
+    ,@(if (eq system-type 'windows-nt) '("-ExecutionPolicy" "Bypass"))
+    "-OutputFormat" "Text"
+    "-File"
+    ,(f-join lsp-pwsh-dir "PowerShellEditorServices/Start-EditorServices.ps1")
+    "-HostName" "'Emacs Host'"
+    "-HostProfileId" "'Emacs.LSP'"
+    "-HostVersion" "0.1"
+    "-LogPath" ,(f-join lsp-pwsh-cache-dir "log.txt")
+    "-LogLevel" "Normal"
+    "-SessionDetailsPath"
+    ,(format "%s/sess-%d.json" lsp-pwsh-cache-dir (incf lsp-pwsh--sess-id))
+    "-AdditionalModules" "@('PowerShellEditorServices.VSCode')"
+    "-Stdio"
+    "-BundledModulesPath" ,lsp-pwsh-dir
+    "-FeatureFlags" "@(' ')"
+    ))
+
+(defun lsp-pwsh--extra-init-params ()
+  "Return form describing parameters for language server."
+  )
+
+(defun lsp-pwsh--force-post-completion (&rest _args)
+  (advice-remove 'company-tng--supress-post-completion 'lsp-pwsh--force-post-completion)
+  nil)
+
+(defvar lsp-pwsh--major-modes '(powershell-mode))
+
+(if (fboundp 'company-lsp)
+    (advice-add 'company-tng-frontend
+                :after
+                #'(lambda (command)
+                    (when (and (eq command 'pre-command)
+                               (memq major-mode lsp-pwsh--major-modes))
+                      (advice-add 'company-tng--supress-post-completion
+                                  :after-while
+                                  'lsp-pwsh--force-post-completion)))))
+
+(lsp-register-client
+ (make-lsp-client
+  :new-connection (lsp-stdio-connection 'lsp-pwsh--command)
+  :major-modes lsp-pwsh--major-modes
+  :server-id 'pwsh-ls
+  :priority 1
+  :initialization-options 'lsp-pwsh--extra-init-params
+  :notification-handlers (lsp-ht ("powerShell/executionStatusChanged" 'ignore))
+  ))
+
+(defun lsp-pwsh--filter-cr (str)
+  "Filter CR entities from STR."
+  (when (and (eq system-type 'windows-nt) str)
+      (replace-regexp-in-string "\r" "" str)))
+(advice-add 'lsp-ui-doc--extract :filter-return #'lsp-pwsh--filter-cr)
+(advice-add 'lsp-ui-sideline--extract-info :filter-return #'lsp-pwsh--filter-cr)
+
+;;; Utils
+(defconst lsp-pwsh-unzip-script "%s -noprofile -noninteractive -nologo -ex bypass -command Expand-Archive -Path '%s' -DestinationPath '%s'"
+  "Powershell script to unzip vscode extension package file.")
+
+(defconst lsp-pwsh-editor-svcs-dl-script "%s -noprofile -noninteractive -nologo -ex bypass -command Invoke-WebRequest -uri '%s' -outfile '%s'"
+  "Command executed via `shell-command' to download the latest PowerShellEditorServices release.")
+
+(defcustom lsp-pwsh-github-asset-url
+  "https://github.com/%s/%s/releases/latest/download/%s"
+  "GitHub latest asset template url."
+  :group 'lsp-pwsh
+  :type 'string)
+
+(defun lsp-pwsh--get-extension (url dest)
+  "Get extension from URL and extract to DEST."
+  (let ((temp-file (concat temporary-file-directory (make-temp-file-internal "ext" nil ".zip" nil))))
```

Comment:
nit: Why don't just use `make-temp-file` with already support to have file expanded in `temporary-file-directory`?
If you have reason to keep this line change, can you also change it to `(make-temp-file-internal "ext" 0 ".zip" nil)` to make sure new file is not created.

Reply:
Good point, think I was testing something, will revert.